### PR TITLE
Attatch the fact to the undefined fact error

### PR DESCRIPTION
**Pull request by chemdrew**

### CODE REVIEW

File: src/errors.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 'use strict'
 
 export class UndefinedFactError extends Error {
-  constructor (...props) {
+  constructor (factId, ...props) {
+    if (!props[0]) props[0] = `Undefined fact: ${factId}`
     super(...props)
```
Comment (CacheControl) [inline src/errors.js]: Let's find a way to ensure error.message is still set to something human readable; we can call ```super(`Undefined fact: ${factId}`)``` here to achieve that.

File: src/errors.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+    this.fact = factId
```
Comment (CacheControl) [inline src/errors.js]: Need a test to assert this value is set

**Comment by CacheControl** [comment]
Closing due to lack of activity; feel free to re-open

**Comment by chemdrew** [comment]
Sorry for disappearing! Use case was for tracing undefined facts to rule executions but realized the ruleResult has all the info for tracing I need so good with keeping this closed out

**Pull request closed without merging**